File: samples/AndroidSamples/salmon-android-samples/src/main/java/com/mku/salmon/samples/MainActivity.java
package com.mku.salmon.samples;

import androidx.appcompat.app.AppCompatActivity;
import androidx.documentfile.provider.DocumentFile;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.provider.DocumentsContract;
import android.widget.Button;
import android.widget.Toast;

import com.mku.android.fs.file.AndroidFile;
import com.mku.android.fs.file.AndroidFileSystem;
import com.mku.fs.file.IFile;
import com.mku.salmon.samples.main.DataActivity;
import com.mku.salmon.samples.main.DataStreamActivity;
import com.mku.salmon.samples.main.FileActivity;
import com.mku.salmon.samples.main.HttpDriveActivity;
import com.mku.salmon.samples.main.LocalDriveActivity;
import com.mku.salmon.samples.main.TextActivity;
import com.mku.salmon.samples.main.WebServiceDriveActivity;

public class MainActivity extends AppCompatActivity {
    private Button textButton;
    private Button dataButton;
    private Button dataStreamButton;
    private Button fileButton;
    private Button localDriveButton;
    private Button httpDriveButton;
    private Button webServiceButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textButton = findViewById(R.id.TEXT_BUTTON);
        textButton.setOnClickListener((e) -> {
            runActivity(TextActivity.class);
        });

        dataButton = findViewById(R.id.DATA_BUTTON);
        dataButton.setOnClickListener((e) -> {
            runActivity(DataActivity.class);
        });

        dataStreamButton = findViewById(R.id.DATA_STREAM_BUTTON);
        dataStreamButton.setOnClickListener((e) -> {
            runActivity(DataStreamActivity.class);
        });

        fileButton = findViewById(R.id.FILE_BUTTON);
        fileButton.setOnClickListener((e) -> {
            runActivity(FileActivity.class);
        });

        localDriveButton = findViewById(R.id.LOCAL_DRIVE_BUTTON);
        localDriveButton.setOnClickListener((e) -> {
            runActivity(LocalDriveActivity.class);
        });

        httpDriveButton = findViewById(R.id.HTTP_DRIVE_BUTTON);
        httpDriveButton.setOnClickListener((e) -> {
            runActivity(HttpDriveActivity.class);
        });

        webServiceButton = findViewById(R.id.WEB_SERVICE_BUTTON);
        webServiceButton.setOnClickListener((e) -> {
            runActivity(WebServiceDriveActivity.class);
        });
    }

    private void runActivity(Class<?> cls) {
        startActivity(new Intent(this, cls));
    }
}

File: samples/DotNetAndroidSamples/DotNetAndroidSamples/MainActivity.cs
using Android.Content;
using AndroidX.AppCompat.App;
using Android.App;
using Android.Widget;
using Android.OS;
using System;
using Mku.Salmon.Samples.Main;

namespace DotNetAndroidSamples;

[Activity(Label = "@string/app_name", MainLauncher = true, Theme = "@style/AppTheme",
    ScreenOrientation = Android.Content.PM.ScreenOrientation.Portrait)]
public class MainActivity : AppCompatActivity
{
    private Button textButton;
    private Button dataButton;
    private Button dataStreamButton;
    private Button fileButton;
    private Button localDriveButton;
    private Button httpDriveButton;
    private Button webServiceButton;

    protected override void OnCreate(Bundle savedInstanceState)
    {
        base.OnCreate(savedInstanceState);

        // Set our view from the "main" layout resource
        SetContentView(Resource.Layout.activity_main);


        textButton = (Button)FindViewById(Resource.Id.TEXT_BUTTON);
        textButton.Click += (s, e) =>
        {
            RunActivity(typeof(TextActivity));
        };

        dataButton = (Button)FindViewById(Resource.Id.DATA_BUTTON);
        dataButton.Click += (s, e) =>
        {
            RunActivity(typeof(DataActivity));
        };

        dataStreamButton = (Button)FindViewById(Resource.Id.DATA_STREAM_BUTTON);
        dataStreamButton.Click += (s, e) =>
        {
            RunActivity(typeof(DataStreamActivity));
        };

        fileButton = (Button)FindViewById(Resource.Id.FILE_BUTTON);
        fileButton.Click += (s, e) =>
        {
            RunActivity(typeof(FileActivity));
        };

        localDriveButton = (Button)FindViewById(Resource.Id.LOCAL_DRIVE_BUTTON);
        localDriveButton.Click += (s, e) =>
        {
            RunActivity(typeof(LocalDriveActivity));
        };

        httpDriveButton = (Button)FindViewById(Resource.Id.HTTP_DRIVE_BUTTON);
        httpDriveButton.Click += (s, e) =>
        {
            RunActivity(typeof(HttpDriveActivity));
        };

        webServiceButton = (Button)FindViewById(Resource.Id.WEB_SERVICE_BUTTON);
        webServiceButton.Click += (s, e) =>
        {
            RunActivity(typeof(WebServiceDriveActivity));
        };

    }

    private void RunActivity(Type cls)
    {
        StartActivity(new Intent(this, cls));
    }
}